ui/src/app/pages/signed-in/component.ts

```diff
@@ -103,6 +112,42 @@ export class SignedInComponent implements OnInit, OnDestroy, AfterViewInit {
     cdrVersionsApi().getCdrVersions().then(resp => {
       cdrVersionStore.next(resp.items);
     });
+
+    this.startUserActivityTracker();
+    this.startInactivityTimers();
+  }
+
+  startUserActivityTracker() {
+    const signalUserActivity = debouncer(() => {
+      window.postMessage(INACTIVITY_CONFIG.MESSAGE_KEY, '*');
+    }, 1000);
+
+    INACTIVITY_CONFIG.TRACKED_EVENTS.forEach(eventName => {
+      window.addEventListener(eventName, () => signalUserActivity(), false);
+    });
+  }
+
+  startInactivityTimers() {
```

Comment:
Could use a comment somewhere around here as to what the rules are of the local storage data, e.g. what it means if the data is unset and how that's handled